File: java/src/main/java/TennisGame7.java
public class TennisGame7 implements TennisGame {
    private final String player1Name;
    private final String player2Name;
    private int player1Score;
    private int player2Score;

    public TennisGame7(String player1Name, String player2Name) {
        this.player1Name = player1Name;
        this.player2Name = player2Name;
    }

    @Override
    public void wonPoint(String playerName) {
        if (playerName.equals("player1"))
            player1Score++;
        else
            player2Score++;

    }

    public String getScore()
    {
        String result = "Current score: ";

        if (player1Score == player2Score)
        {
            // tie score
            switch (player1Score)
            {
                case 0:
                    result += "Love-All";
                    break;
                case 1:
                    result += "Fifteen-All";
                    break;
                case 2:
                    result += "Thirty-All";
                    break;
                default:
                    result += "Deuce";
                    break;
            }
        }
        else if (player1Score >= 4 || player2Score >= 4)
        {
            // end-game score
            if (player1Score - player2Score == 1) {
                result += "Advantage " + player1Name;
            } else if (player1Score - player2Score == -1) {
                result += "Advantage " + player2Name;
            } else if (player1Score - player2Score >= 2) {
                result += "Win for " + player1Name;
            } else {
                result += "Win for " + player2Name;
            }
        }
        else
        {
            // regular score
            result +=  switch (player1Score)
            {
                case 0 -> "Love";
                case 1 -> "Fifteen";
                case 2 -> "Thirty";
                default -> "Forty";
            };
            result += "-";
            result +=  switch (player2Score)
            {
                case 0 -> "Love";
                case 1 -> "Fifteen";
                case 2 -> "Thirty";
                default -> "Forty";
            };

        }

        return result + ", enjoy your game!";
    }
}

File: csharp/Tennis/TennisGame7.cs
namespace Tennis;

public class TennisGame7 : ITennisGame
{
    private int player1Score;
    private int player2Score;
    private string player1Name;
    private string player2Name;

    public TennisGame7(string player1Name, string player2Name)
    {
        this.player1Name = player1Name;
        this.player2Name = player2Name;
    }

    public void WonPoint(string playerName)
    {
        if (playerName == player1Name)
            player1Score++;
        else
            player2Score++;
    }

    public string GetScore()
    {
        string result = "Current score: ";

        if (player1Score == player2Score)
        {
            // tie score
            switch (player1Score)
            {
                case 0:
                    result += "Love-All";
                    break;
                case 1:
                    result += "Fifteen-All";
                    break;
                case 2:
                    result += "Thirty-All";
                    break;
                default:
                    result += "Deuce";
                    break;
            }
        }
        else if (player1Score >= 4 || player2Score >= 4)
        {
            // end-game score
            switch (player1Score - player2Score)
            {
                case 1:
                    result += $"Advantage {player1Name}";
                    break;
                case -1:
                    result += $"Advantage {player2Name}";
                    break;
                case >= 2:
                    result += $"Win for {player1Name}";
                    break;
                default:
                    result += $"Win for {player2Name}";
                    break;
            }
        }
        else
        {
            // regular score
            result += player1Score switch
            {
                0 => "Love",
                1 => "Fifteen",
                2 => "Thirty",
                _ => "Forty"
            };

            result += "-";

            result += player2Score switch
            {
                0 => "Love",
                1 => "Fifteen",
                2 => "Thirty",
                _ => "Forty"
            };
        }

        return result + ", enjoy your game!";
    }
}